game_event/tests.py

```diff
@@ -252,3 +273,48 @@ def test_fail_add_player_to_event_already_in_it(self, client, game_event, player
         assert response.status_code == 200
         with pytest.raises(IntegrityError):
             response = client.post(url, {'answer': 'yes'})
+
+
+@pytest.mark.django_db
+class TestGetWeatherFromApi:
+    def test_one_valid_response(self, game_event, player, mock_urlopen):
+        mock_response = mock.Mock()
+        mock_response.status = 200
+        mock_response.read.return_value = bytes('{"weather": [{"icon": "04d"}],"main": {"temp": 13.19}}', 'utf-8')
+        mock_urlopen.return_value = mock_response
+        output = get_weather_from_api([game_event], mock_urlopen)
+        game_event.weather = {'icon': "http://openweathermap.org/img/wn/04d@2x.png", 'temp': 13.19}
+        assert output == [game_event]
+
+    def test_exception_thrown_from_api_call(self, game_event, player, mock_urlopen):
+
+        mock_urlopen.side_effect = Exception()
+
+        output = get_weather_from_api([game_event], mock_urlopen)
+        assert output[0].weather is None
```

Comment:
My comment remains regarding asserting the entire collection rather just the one element.

Reply:
_"In that case, you can fall back to asserting on the weather directly rather than the complete list. We tried testing more thoroughly. Didn't work"_

After that. Is the assertion meets you expectations? @yftacherzog

Reply:
Yes. That should do.